crates/fluvio-sc/src/controllers/topics/controller.rs

```diff
@@ -152,4 +164,34 @@ impl<C: MetadataItem> TopicController<C> {
             }
         }
     }
+
+    async fn ensure_offsets_topic_exists(&mut self) {
```

Comment:
Hmm.  This is not correct usage of TopicController.   The purpose of TopicController is handle topic lifecycle such as topic->partitioning mapping.  This assumes Topic object is created by external entity not the controller.    So to follow similar pattern, we should have something like `SystemTopicController` which will ensure system topic exists. 

However, not sure use of controller is right approach.   We could instead
- Create/update system topic when SC is started.   
- Don't allow system topic to be deleted.

Reply:
I created `SystemTopicController` and moved the logic into it.

Regarding the topic deletion, it is not a problem to forbid deletion, but what if it does need to be deleted for some reason? Troubleshooting, dev purposes, memory problems, etc. We need to provide at least some way of deleting, if not the topic but the content maybe. So, I would propose postponing that thing for later, until we figure out a convenient strategy for system topics.